catalogue_pipeline/transformer/src/main/scala/uk/ac/wellcome/transformer/transformers/TransformableTransformer.scala

```diff
@@ -3,15 +3,16 @@ package uk.ac.wellcome.transformer.transformers
 import uk.ac.wellcome.models.UnidentifiedWork
 import uk.ac.wellcome.models.transformable.Transformable
 
+import scala.reflect.ClassTag
 import scala.util.Try
 
-trait TransformableTransformer[+T <: Transformable] {
+trait TransformableTransformer[T <: Transformable] {
   protected[this] def transformForType(
     t: T,
     version: Int): Try[Option[UnidentifiedWork]]
 
-  def transform(transformable: Transformable,
-                version: Int): Try[Option[UnidentifiedWork]] =
+  def transform(transformable: Transformable, version: Int)(
+    implicit tag: ClassTag[T]): Try[Option[UnidentifiedWork]] =
```

Comment:
I'd really avoid reflection stuff here. I have a fix for this using partial functions that I have meant to do for a while